presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java

```diff
@@ -174,15 +178,20 @@ private DistributedQueryRunner(
             extraCoordinatorProperties.put("experimental.iterative-optimizer-enabled", "true");
             extraCoordinatorProperties.putAll(extraProperties);
             extraCoordinatorProperties.putAll(coordinatorProperties);
-            coordinator = closer.register(createTestingPrestoServer(discoveryUrl, false, resourceManagerEnabled, true, extraCoordinatorProperties, parserOptions, environment, baseDataDir));
-            servers.add(coordinator);
+            for (int i = 0; i < coordinatorCount; i++) {
+                TestingPrestoServer coordinator = closer.register(createTestingPrestoServer(discoveryUrl, false, resourceManagerEnabled, true, extraCoordinatorProperties, parserOptions, environment, baseDataDir));
+                servers.add(coordinator);
+                coordinators.add(coordinator);
+                extraCoordinatorProperties.remove("http-server.http.port");
```

Comment:
Not sure if we have verified it or not but with multiple coordinator configuration the port is going to be shared and so it will fail. At max it will allow two coordinator given one coordinator port set using this property, and ther other one is default which is 0. We should make necessary changes to support multi coordinator as part of the distributed query runner.